src/gc/gcinterface.h

```diff
@@ -174,6 +181,9 @@ enum end_no_gc_region_status
 
 typedef BOOL (* walk_fn)(Object*, void*);
 typedef void (* gen_walk_fn)(void* context, int generation, uint8_t* range_start, uint8_t* range_end, uint8_t* range_reserved);
+typedef void (* record_surv_fn)(uint8_t* begin, uint8_t* end, ptrdiff_t reloc, size_t context, BOOL compacting_p, BOOL bgc_p);
```

Comment:
use c++ `bool` here and below instead of `BOOL`?

Also - after moving things to the VM side, do you think we still need `ScanContext` and `ProfilingScanContext` to be on the GC interface anymore?
Also - after moving things to the VM side, do you think we still need `ScanContext` and `ProfilingScanContext` on the GC interface anymore?